clients/go/gengo.go

```diff
@@ -381,10 +381,15 @@ func (c *WagClient) do%sRequest(ctx context.Context, req *http.Request, headers
 	}
 	resp, err := c.requestDoer.Do(client, req)
 	if err != nil {
+		c.logger.ErrorD("client-request-finished", map[string]string{
```

Comment:
I'm hoping we can in the future add this in non error conditions too, but I think we could add the `error` in the signalfx series at the matcher level. 

Yes. Thanks for pinging - it is a bug: 

TIL: ```A non-2xx status code doesn't cause an
  // error.```
https://golang.org/src/net/http/client.go?s=16536:16588#L463